tests/flask_app/__init__.py

```diff
@@ -38,7 +39,12 @@ def bad_url():
 
     @app.route("/redirect/")
     def redirect_to_index():
-        return redirect(url_for("index"))
+        code = request.args.get('code')
+
+        if code:
+            return redirect(url_for("index"), code=code)
```

Comment:
Could change this to `code = request.args.get('code') or '301'` and then collapse the if block